libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,144 @@
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        /// <param name="storage">The <see cref="IStorage"/> implementation to use for this <see cref="Dialog"/>.</param>
+        /// <param name="botFrameworkAuthentication">A <see cref="BotFrameworkAuthentication"/> used to obtain a client for making calls to Bot Builder Skills.</param>
+        public AdaptiveDialogBot(
+            ResourceExplorer resourceExplorer,
+            string rootDialogId,
+            string defaultLocale,
+            ILogger<AdaptiveDialogBot> logger = null,
+            IStorage storage = null,
+            BotFrameworkAuthentication botFrameworkAuthentication = null)
+            : this(
+                  resourceExplorer,
+                  rootDialogId ?? DefaultDialogName,
+                  defaultLocale ?? DefaultLocale,
+                  logger ?? NullLogger<AdaptiveDialogBot>.Instance,
+                  botFrameworkAuthentication ?? BotFrameworkAuthenticationFactory.Create(),
+                  new SkillConversationIdFactory(storage ?? new MemoryStorage()),
+                  new ConversationState(storage ?? new MemoryStorage()),
+                  new UserState(storage ?? new MemoryStorage()))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogBot"/> class.
+        /// </summary>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="rootDialogId">The id of the dialog to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        /// <param name="botFrameworkAuthentication">A <see cref="BotFrameworkAuthentication"/> used to obtain a client for making calls to Bot Builder Skills.</param>
+        /// <param name="skillConversationIdFactoryBase">A <see cref="SkillConversationIdFactoryBase"/> implementation.</param>
+        /// <param name="conversationState">A <see cref="ConversationState"/> implementation.</param>
+        /// <param name="userState">A <see cref="UserState"/> implementation.</param>
+        public AdaptiveDialogBot(
+            ResourceExplorer resourceExplorer,
+            string rootDialogId,
+            string defaultLocale,
+            ILogger<AdaptiveDialogBot> logger,
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            SkillConversationIdFactoryBase skillConversationIdFactoryBase,
+            ConversationState conversationState,
+            UserState userState)
+        {
+            _resourceExplorer = resourceExplorer;
+            _rootDialogId = rootDialogId ?? "Main.dialog";
```

Comment:
We already provide a default that uses the constant in the other constructor, make this parameter required in this constructor.